### PR TITLE
feat(fecth): corsCheck

### PR DESCRIPTION
**Pull request by hemanth**
CORS check.

### CODE REVIEW

File: lib/fetch/util.js [hunk 1]
```diff
@@ -168,9 +168,38 @@ function crossOriginResourcePolicyCheck () {
...
+
+  // 3. If request’s credentials mode is not "include" and origin is `*`, then return success.
+  if (request.credentials !== 'include' && allowedOrigin === '*') {
+    return 'success'
+  }
+
+  // 4. If the result of byte-serializing a request origin with request is not origin, then return failure.
+  // TODO: not very clear what byte-serializing means.
+  // https://fetch.spec.whatwg.org/#byte-serializing-a-request-origin
+  // Should we do the isomorphic encoding?
```
Comment (hemanth) [inline lib/fetch/util.js:188]: Seeking some clarity on this.

**Review by mcollina — COMMENTED** [review]
Why are we adding this? We'd
never support cors.

**Comment by ronag** [comment]
@benjamingr

**Comment by KhafraDev** [comment]
If this PR isn't merged 83defac59ca50494e88b10ff22226b524f1f7dee should probably get reverted too as it deals with cross origin stuff as well

> Cross-origin read blocking, better known as CORB, is an algorithm which identifies dubious cross-origin resource fetches (e.g., fetches that would fail anyway like attempts to render JSON inside an img element) and blocks them before they reach a web page. CORB reduces the risk of leaking sensitive data by keeping it further from cross-origin web pages.

**Comment by benjamingr** [comment]
@ronag I agree with Matteo on this

**Comment by ronag** [comment]
We should replace the implementation parts with a clear comment that this stub is out of scope.

**Comment by hemanth** [comment]
Ah, it makes more sense, was doubtful too. 

Should we revisit the other unimplemented methods?

**Comment by hemanth** [comment]
We have the following to be implemented:

1. crossOriginResourcePolicyCheck
2. TAOCheck
3. coarsenedSharedCurrentTime
4. makePolicyContainer
5. clonePolicyContainer
6. determineRequestsReferrer
7. tryUpgradeRequestToAPotentiallyTrustworthyURL

**Review by mcollina — CHANGES_REQUESTED** [review]
We should be checking CORS in fetch() in Undici.

Could you please document this divergence from the spec instead?

**Comment by mcollina** [comment]
We are not going to support CORS in Node.js.

**Pull request closed without merging**